mobx/lib/src/interceptable.dart

```diff
@@ -8,8 +8,7 @@ abstract class Interceptable<T> {
   Dispose intercept(Interceptor<T> interceptor);
 }
 
-class Interceptors<T>
-    extends NotificationHandlers<WillChangeNotification<T>, Interceptor<T>>
+class Interceptors<T> extends NotificationHandlers<WillChangeNotification<T>>
```

Comment:
It should probably override `add` to take a `Interceptor<T>`, to make things type safe

Reply:
Yup, great observation 👍

Reply:
Added the checks and tests for it. Please review @rrousselGit

Reply:
I would expect using `covariant` keyword instead:

```dart
class NotificationHandlers<TNotification> {
  VoidCallback add(covariant Function handler) {}
}


class Foo extends NotificationHandlers<Object> {
  @override
  VoidCallback add(ValueChanged<Object> handler) => super.add(handler);
}
```

This makes the change safe at compile time

Reply:
Ah missed this ....not sure how, but this comment did not show up earlier. Will do this change.